lib/Dialect/TTNN/IR/TTNNOpsAttrs.cpp

```diff
@@ -34,6 +39,16 @@ bool TTNNLayoutAttr::isTiled() const {
   return ::mlir::isa<::mlir::tt::TileType>(getElementType());
 }
 
+// Check if the tensor memory layout is in L1 memory
+bool TTNNLayoutAttr::hasL1TensorMemoryLayout() const {
```

Comment:
We do have problems with conflicting naming with tensor related stuff in general. But doesn't tensor memory layout refer to interleaved / sharded, does it make more sense to call this hasL1BufferType?